extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSink.java

```diff
@@ -89,11 +93,19 @@ protected StreamResult<Object> transferParts(List<DataSource.Part> parts) {
         return StreamResult.success();
     }
 
-    private String getDestinationObjectName(String partName) {
+    private String getDestinationObjectName(String partName, int partsSize) {
+        if (isNullOrEmpty(objectName) && !isNullOrEmpty(keyName)) {
+            monitor.warning("The usage of the property \"keyName\" to define the object name is deprecated" +
+                    " since version 0.5.2. Please use the \"objectName\" property instead to name objects in " +
+                    "destination.");
+            this.objectName = keyName;
+        }
+        var name = (partsSize == 1 && !StringUtils.isNullOrEmpty(objectName)) ? objectName : partName;
```

Comment:
the way the name is built is not tested appropriately (if I make this method return either `objectName` or `partName`, all tests contained in `S3DataSinkTest` will pass)